src/cls/IPM/Repo/Oras/Definition.cls

```diff
@@ -0,0 +1,183 @@
+
+Property URL As %String(MAXLEN = 2048) [ Required ];
+
+Index ORASURL On URL [ Unique ];
+
+Parameter URLPromptString = {$$$Text("ORAS Registry URL:","ZPM")};
+
+Property Namespace As %String;
+
+Property Username As %String(MAXLEN = "");
+
+Property Password As %String(MAXLEN = "");
+
+Property Token As %String(MAXLEN = "");
+
+Property DeploymentEnabled As %Boolean [ InitialExpression = 0 ];
+
+Index DeploymentServerOras On DeploymentEnabled [ Unique ];
+
+Property Details As %String(MAXLEN = "") [ SqlComputeCode = {Set {*} = {URL}}, SqlComputed, SqlComputeOnChange = (%%INSERT, %%UPDATE) ];
+
+XData Commands [ XMLNamespace = "http://www.intersystems.com/PackageManager/CLI" ]
+{
+<commands>
+<command name="repo">
+<group name="Oras repository">
+
+<modifier name="namespace" aliases="ns" value="true" description="For ORAS repositories, specifies the registry namespace, i.e. the fully specified path to a package is &lt;url>/&lt;namespace>/&lt;package>.
+	Note, this is not the IRIS namespace, but a namespace in the sense of a (sub)directory in the registry's storage system. For example, if in a custom managed registry, the base url is 'registry.company.com',
+	and the package is 'objectscript-example', but it will be stored at 'registry.company.com/packages/ipm/objectscript-example', then the namespace will be 'packages/ipm'." />
+</group>
+</command>
+</commands>
+}
+
+/// This callback method is invoked by the <METHOD>%Save</METHOD> method to 
+/// provide notification that the object is being saved. It is called before 
+/// any data is written to disk.
+/// 
+/// <P><VAR>insert</VAR> will be set to 1 if this object is being saved for the first time.
+/// 
+/// <P>If this method returns an error then the call to <METHOD>%Save</METHOD> will fail.
+Method %OnBeforeSave(insert As %Boolean) As %Status [ Private, ServerOnly = 1 ]
+{
+	// Leave DeploymentEnabled null to make unique index work right
+	// (i.e., deployment can only be enabled for one remote server.)
+	Set:..DeploymentEnabled=0 ..DeploymentEnabled=""
+	Quit $$$OK
+}
+
+/// Returns a REST client to interact with this server.
+Method GetPackageService() As %IPM.Repo.IPackageService
+{
+	Set tClient = ##class(%IPM.Repo.Oras.PackageService).%New()
+	Set tClient.Location = ..URL
+	Set tClient.Namespace = ..Namespace
+	Set tClient.Username = ..Username
+	Set tClient.Password = ..Password
+    Set tClient.Token = ..Token
+	Quit tClient
+}
+
+/// Returns a REST client to publish modules/applications to this server.
+Method GetPublishService() As %IPM.Repo.IPublishService
+{
+	Set tClient = ##class(%IPM.Repo.Oras.PublishService).%New()
+	Set tClient.Location = ..URL
+	Set tClient.Namespace = ..Namespace
+	Set tClient.Username = ..Username
+	Set tClient.Password = ..Password
+    Set tClient.Token = ..Token
+	Quit tClient
+}
+
+/// Outputs information about this server to the current device.
+/// Subclasses may override to show additional information.
+Method Display()
```

Comment:
A lot of the logic here is similar to that of the existing remote repo so could do some refactoring into helper methods. I think this would be true for not just ORAS but any other remote repositories we add support to in future

Reply:
Refactored into superclass.